pywwt/core.py

```diff
@@ -220,14 +228,22 @@ def set_current_time(self, dt=None):
             astropy :class:`astropy.time.Time` object. If not specified, this
             uses the current time
         """
+        # Ensure the object received is a datetime or Time; convert it to UTC
         if dt is None:
-            dt = Time.now()
-        if isinstance(dt, Time):
-            dt = dt.datetime
-        self._send_msg(event='set_datetime',
-                       year=dt.year, month=dt.month, day=dt.day,
-                       hour=dt.hour, minute=dt.minute, second=dt.second,
-                       millisecond=int(dt.microsecond / 1000.))
+            utc_dt = datetime.utcnow().astimezone(pytz.UTC).isoformat()
+        elif isinstance(dt, datetime):
+            if dt.tzinfo is None:
+                utc_dt = pytz.utc.localize(dt).isoformat()
+            elif dt.tzinfo == pytz.UTC:
+                utc_dt = dt.isoformat()
+            else:  # has a non-UTC time zone
+                utc_dt = dt.astimezone(pytz.UTC).isoformat()
+        elif isinstance(dt, Time):
+            utc_dt = dt.to_datetime(pytz.UTC).isoformat()
+        else:
+            raise ValueError('Time must be a datetime or astropy.Time object')
```

Comment:
If I'm reading correctly, very similar code appears later; it should be centralized into a utility function.